src/openrct2/EditorObjectSelectionSession.cpp

```diff
@@ -499,12 +499,15 @@ static void set_object_selection_error(uint8_t is_master_object, rct_string_id e
  *
  *  rct2: 0x006AB54F
  */
-bool window_editor_object_selection_select_object(uint8_t isMasterObject, int32_t flags, const ObjectRepositoryItem* item)
+GameActions::Result window_editor_object_selection_select_object(
+    uint8_t isMasterObject, int32_t flags, const ObjectRepositoryItem* item)
 {
+    auto res = GameActions::Result();
+
     if (item == nullptr)
     {
         set_object_selection_error(isMasterObject, STR_OBJECT_SELECTION_ERR_OBJECT_DATA_NOT_FOUND);
-        return false;
```

Comment:
You still need to modify set_object_selection_error so that the error message is returned via the result